packages/remix-dev/__tests__/routesConvention-test.ts

```diff
@@ -36,6 +36,47 @@ describe("createRoutePath", () => {
       ["[index]", "index"],
       ["test/inde[x]", "test/index"],
       ["[i]ndex/[[].[[]]", "index/[/[]"],
+
+      // Optional segment routes
+      ["(routes)/$", "routes?/*"],
+      ["(routes)/($)", "routes?/*?"], // TODO: Fails, do we want to allow this?
```

Comment:
I don't think we need to support optional splats as they're already optional.  `path="child/*"` will match both `/child` and `/child/any/other/path`:  See https://codesandbox.io/s/optional-splat-segments-xpn3kq

Maybe we throw an error on `($)` syntax?